src/backend/distributed/sql/udfs/fix_partition_shard_index_names/10.2-1.sql

```diff
@@ -0,0 +1,29 @@
+CREATE FUNCTION pg_catalog.fix_partition_shard_index_names(table_name regclass)
+  RETURNS void
+  LANGUAGE C STRICT
+  AS 'MODULE_PATHNAME', $$fix_partition_shard_index_names$$;
+COMMENT ON FUNCTION pg_catalog.fix_partition_shard_index_names(table_name regclass)
+  IS 'fix index names on partition shards';
+
+CREATE OR REPLACE FUNCTION pg_catalog.fix_partition_shard_index_names()
+  RETURNS SETOF regclass
+  LANGUAGE plpgsql
+  AS $$
+DECLARE
+	oid regclass;
+BEGIN
+    FOR oid IN SELECT c.oid
+            FROM pg_dist_partition p
+            JOIN pg_class c ON p.logicalrelid = c.oid
+			JOIN pg_namespace n ON c.relnamespace = n.oid
```

Comment:
why do we need the join on the `pg_namespace`, isn't oid unique?

Reply:
You are right, it's not needed.